src/utils/system_utils/src/system_utils.c

```diff
@@ -67,6 +68,52 @@ const char* ADUC_SystemUtils_GetTemporaryPathName()
     return env;
 }
 
+/**
+ * @brief mktemp() implementation
+ *
+ * mktemp() is generally considered unsafe as the temporary names can be easy to guess, and there are
+ * a limited number of combinations.
+ *
+ * However, for unit tests, it can be useful, so it's highly recommended that this method is only
+ * used for unit tests.
+ *
+ * @param tmpl The last six characters XXXXXX are replaced with a string that makes the filename unique.
+ * @return Returns @param tmpl.
+ */
+char* ADUC_SystemUtils_MkTemp(char* tmpl)
+{
+    const size_t len = strlen(tmpl);
+    if (len < 6)
+    {
+        tmpl[0] = '\0';
+        errno = EINVAL;
+        return tmpl;
+    }
+
+    // Start of XXXXXX
+    char* Xs = tmpl + (len - 6);
+
+    // Get a randomish number from real time clock.
```

Comment:
why not use srand()/rand()?  Don't see a compelling reason to roll one's own PRNG.  If for production use, it could use /dev/random or /dev/urandom on *nix and {Rtl,Crypt}GenRandom on win32, or implement in c++ using <random>

Reply:
i started with that and but rand likes to return values like 1000 and 23, which isn't enough digits for my needs. Maybe I was doing something wrong?